packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -1,631 +1,339 @@
-        self[i] = { component: menuItem };
-      }
-    });
-
-    return contextMenuItems;
+  toogleSearch() {
+    this.searchDialogOpened = !this.searchDialogOpened;
   }
 
-  /**
-   * Highlight menu item with special class.
-   * Improves visual "Where am I?" navigation clarity.
-   *
-   * @since 2020.04.12
-   * @private
-   */
-  _highlightCurrentMenuItem() {
-    this.menuItemsElements.forEach((menuItemsEl) => {
-      const currentMenuItemEl = menuItemsEl.querySelector('.current-menu-item');
-
-      if (currentMenuItemEl && currentMenuItemEl.id && menuItemsEl.items) {
-        const idx = menuItemsEl.items.findIndex((i) => i.id === currentMenuItemEl.id);
-
-        menuItemsEl.setAttribute('selected', idx);
-      }
-    });
+  // eslint-disable-next-line class-methods-use-this
+  _searchDialogRenderer() {
+    return html` <div class="search-form-container"></div> `;
   }
 
-  /**
-   * Populate children `<vaadin-context-menu>` elements.
-   *
-   * @private
-   * @todo Links cannot be sub-menu hosts?
-   */
-  _updatedContextMenuItems() {
-    Object.values(this.contextMenuItems).forEach((items) => {
-      items.forEach((menuItem) => {
-        if (!menuItem.children || !menuItem.children.length) {
-          return;
-        }
-
-        const contextMenu = this.querySelector(
-          `vaadin-tab#menu-item-${menuItem.id} > vaadin-context-menu`
-        );
-
-        // Populate.
-        contextMenu.items = this._createContextMenuItems(menuItem.children);
-
-        // Prevent close on upstream events: clicks, keydown, etc
-        contextMenu.addEventListener('item-selected', (e) => {
-          e.stopImmediatePropagation();
-        });
-      });
-    });
+  // eslint-disable-next-line class-methods-use-this
+  _onBackBtnClick(e) {
+    e.stopImmediatePropagation();
+    [...document.body.querySelectorAll('vaadin-context-menu-overlay')].at(-1).close();
   }
 
-  /**
-   * Re-orient menu items.
-   *
-   * @see https://github.com/vaadin/vaadin-context-menu/blob/v4.3.12/src/vaadin-device-detector.html#L12
-   * @see https://github.com/vaadin/vaadin-context-menu/issues/253
-   * @todo refactor w/ `<cxl-icon-nav>`?
-   */
-  _reorientMenuItems() {
-    let orientation = 'vertical';
-
-    if (this.wide) {
-      orientation = 'horizontal';
+  // Creates and returns a custom menu item component for use in vaadin-menu-bar
+  createItem(
+    { text, description, sectionheader, component, icon, href, children, depth, split },
+    group
+  ) {
+    // If divider, return an <hr> element, nothing else
+    if (component === 'hr') {
+      return document.createElement('hr');
     }
 
-    this.menuItemsElements.forEach((el) => {
-      el.setAttribute('orientation', orientation);
-      el.setAttribute('wide', this.wide);
-    });
+    // create menu item
+    const menuItemElement = document.createElement('vaadin-context-menu-item');
+    if (group) menuItemElement.classList.add(`${group}-menu-item`);
```

Comment:
Move it below the "classes" comment.

```suggestion
    if (group) {
        menuItemElement.classList.add(`${group}-menu-item`);
    }
```